components/contribution-flow/ContributeProfilePicker.js

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { groupBy, sortBy, truncate } from 'lodash';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import { CollectiveType } from '../../lib/constants/collectives';
+
+import Avatar from '../Avatar';
+import CollectivePicker, { FLAG_COLLECTIVE_PICKER_COLLECTIVE, FLAG_NEW_COLLECTIVE } from '../CollectivePicker';
+import { Flex } from '../Grid';
+import { Span } from '../Text';
+const { USER, ORGANIZATION, COLLECTIVE, FUND, EVENT, PROJECT } = CollectiveType;
+
+const filterProfiles = (profiles, account, canUseIncognito) => {
+  return profiles.filter(p => {
+    // if admin of collective you are donating to, remove it from the list
+    if (p.id === account.legacyId) {
+      return false;
+    } else if (!canUseIncognito && p.isIncognito) {
+      return false;
+    } else if (p.type === 'COLLECTIVE' && (!p.host || p.host.id !== account.host?.legacyId)) {
+      return false;
+    } else {
+      return true;
+    }
+  });
+};
+
+const formatAccountName = (intl, account) => {
+  return account.isIncognito
+    ? intl.formatMessage({ id: 'profile.incognito', defaultMessage: 'Incognito' })
+    : truncate(account.name, { length: 40 });
+};
+
+const getProfileOptions = (intl, profiles, canUseIncognito) => {
+  const getOptionFromAccount = value => ({ [FLAG_COLLECTIVE_PICKER_COLLECTIVE]: true, value, label: value.name });
+  const sortOptions = options => sortBy(options, 'value.name');
+  const profileOptions = profiles.map(getOptionFromAccount);
+  const profilesByType = groupBy(profileOptions, p => p.value.type);
+  const myself = profilesByType[USER] || [];
+  const myOrganizations = sortOptions(profilesByType[ORGANIZATION] || []);
+
+  // Add incognito profile entry if it doesn't exists
+  if (canUseIncognito) {
+    const hasIncognitoProfile = profiles.some(p => p.type === 'USER' && p.isIncognito);
+    if (!hasIncognitoProfile) {
+      // TODO Test this with a new user
+      myself.push(
+        getOptionFromAccount({
+          id: 'incognito',
+          type: 'USER',
+          isIncognito: true,
+          name: intl.formatMessage({ id: 'profile.incognito', defaultMessage: 'Incognito' }),
```

Comment:
Currently we show something like this; 

![image](https://user-images.githubusercontent.com/12435965/146798835-7ec66315-e3ef-41b1-9cfc-09802d61c80e.png)

Note that it says, `Check privacy box for info`. This doesn't seem to make sense I think. If what's meant by privacy box is the blue box beneath the collective picker, it doesn't mention anything about incognito contributions. The FAQ on the right side does mention about incognito contributions, maybe we can change the line to say, "Check FAQ for more info"; wdyt? 🤔

Reply:
Good catch, it doesn't make much sense ideed. I'm going to rephrase this whole thing